src/stats/probs/cdf.rs

```diff
@@ -138,12 +142,15 @@ pub struct CDF<T: Ord> {
 }
 
 impl<T: Ord> CDF<T> {
-    /// Create CDF from given probability mass function (PMF). The PMF may contain duplicate values
-    /// the probabilities of which are summed during generation of the CDF.
+    /// Create CDF from a vector representing a probability mass function (PMF).
+    /// The PMF may contain duplicate values the probabilities of which are summed
+    /// during generation of the CDF.
+    ///
+    /// Runtime complexity: O(n), where n is the number of `entries`.
```

Comment:
```suggestion
    /// Runtime complexity: O(n log n), where n is the number of `entries`.
```
(since values get sorted first, assuming  `cmp` is O(1))

Reply:
OK, the sort may be `O(log n)`. But the function then `for`-loops through the sorted vector and adds the probability onto the previous summed probability. If I interpret this correctly, that means that work is done for every entry, and thus the overall runtime complexity should scale with `n`, not `log n`, right?

The comparison is only there, to check whether to stick the new sum into a new entry or whether you had entries with the same `value` in the probability mass function and thus want to stick the new sum into the already existing last entry.

Reply:
Sorting is n log n already

Reply:
Ah, `O(n log n)`, not `O(log n)`. What's the English equivalent of `Wer lesen kann, ist klar im Vorteil!`.